### PR TITLE
Comments autolink

### PR DESCRIPTION
**Pull request by bago2k4**
Card: https://trello.com/c/LyZOf6zw

From a discussion in Slack with Stuart we decided to adopt MediumEditor even for our new and edit comment fields. This means that we can use its autolink feature that automatically turn links into anchor and the paste one that clean all the rich HTML before pasting content in the field.
All the rest of the features are disabled.

To test:
- focus the add comment link
- press Escape: do you NOT see the modal dismiss and go back to the board view? Good
- focus on the add comment and blur it while the field is empty, does it collapse back? Good
- type something with a url in it and check that it's turned into a link, it works? Good
- select some text, do you NOT see the MediumEditor toolbar appearing? Good
- while typing use CMD+B or CMD+I or CMD+U, do you NOT see the bold, italic or underline formatting appear when you type more? Good
- add the comment, does the link is still clickable in the new added comment? Good
- now from the horizontal ellipsis of the newly added comment click edit
- press Escape while the comment is in edit mode, do you NOT see the modal dismiss and go back to board? Good
- edit again, now click outside the comment, is it NOT the comment still editable? Good
- repeat the above tests of the add comment for the comment edit

NB: archive this card https://trello.com/c/BZ0fA4Rm/1113-auto-link-urls-in-comments when done since it's a duplicate of this one but with less details

### CODE REVIEW

**Comment by bago2k4** [comment]
[Comments autolink and paste cleanup.](https://trello.com/c/LyZOf6zw/1139-comments-autolink-and-paste-cleanup)

File: src/oc/web/components/comments.cljs [hunk 1]
```diff
@@ -19,39 +19,112 @@
...
+      (reset! (::esc-key-listener s) nil))
+    (reset! (::medium-editor s) nil)
+    (reset! (::editing? s) false)
+    (dis/dispatch! [:input [:comment-edit] nil])))
+
+(defn cancel-edit
+  [e s c]
+  (.stopPropagation e)
+  (let [comment-field (rum/ref-node s "comment-body")]
+    (set! (.-innerHTML comment-field) (utils/emojify @(::initial-body s) true))
```
Comment (thepug) [inline src/oc/web/components/comments.cljs]: I am seeing a bug and I don't think you want `::initial-body` here. To recreate the issue edit a comment and save with enter.  After that open the comment and hit escape. You will see the original data.

What you want here is to use the current comment data. This is where you will want to read data from the "Store" to get the current state. Right now that means read from the app state.

Comment (bago2k4) [reply src/oc/web/components/comments.cljs]: @thepug addressed the issue, good catch!

**Comment by thepug** [comment]
One more issue.

Open a post.  Click the "Add comment" box.  Click escape.  Click the "Add comment" box again and you are unable to post a comment.

**Comment by thepug** [comment]
Now I don't get the 'Add' button after typing. 

http://recordit.co/m4eMtmLGyq
m4emtmlgyq

**Comment by bago2k4** [comment]
@thepug good catch again, my bad i introduced a regression while fixing the last issue you found. Now it should be good

**Comment by thepug** [comment]
> @thepug good catch again, my bad i introduced a regression while fixing the last issue you found. Now it should be good

One more item has to do with empty comments.

oyvuaw8xjy

**Comment by bago2k4** [comment]
@thepug addressed

**Comment by thepug** [comment]
Man, I hate that I am catching these one at a time but each change makes me try testing it differently.

An issue when changing the comment to blank while editing.

http://recordit.co/QZMyFNFBFG
qzmyfnfbfg

**Pull request merged**